libcontainer/cgroups/fs2/memory_test.go

```diff
@@ -0,0 +1,349 @@
+		t.Fatal("Got the wrong value, set memory.soft_limit_in_bytes failed.")
+	}
+}
+
+func TestMemorySetMemoryswapV2(t *testing.T) {
+	cgroups.TestMode = true
+
+	fakeCgroupDir := t.TempDir()
+	maxPath := filepath.Join(fakeCgroupDir, "memory.max")
+	lowPath := filepath.Join(fakeCgroupDir, "memory.low")
+	swapMaxPath := filepath.Join(fakeCgroupDir, "memory.swap.max")
+
+	const (
+		memoryswapBefore = 314572800 // 300M
+		memoryswapAfter  = 524288000 // 500M
+	)
+	if err := os.WriteFile(maxPath, []byte(strconv.Itoa(1)), 0o644); err != nil {
+		t.Fatal(err)
+	}
+	if err := os.WriteFile(lowPath, []byte(strconv.Itoa(1)), 0o644); err != nil {
+		t.Fatal(err)
+	}
+	if err := os.WriteFile(swapMaxPath, []byte(strconv.Itoa(memoryswapBefore)), 0o644); err != nil {
+		t.Fatal(err)
+	}
+
+	r := &configs.Resources{
+		Memory:            1,
+		MemoryReservation: 1,
+		MemorySwap:        memoryswapAfter,
+	}
+	if err := setMemory(fakeCgroupDir, r); err != nil {
+		t.Fatal(err)
+	}
+
+	value, err := fscommon.GetCgroupParamUint(fakeCgroupDir, "memory.swap.max")
+	if err != nil {
+		t.Error(err)
+	}
+	if value != memoryswapAfter-1 {
+		t.Fatalf("Got the wrong value %d, set memory.swap.max failed.", value)
+	}
+}
+
+func TestMemoryStats(t *testing.T) {
+	cgroups.TestMode = true
+
+	fakeCgroupDir := t.TempDir()
+	statPath := filepath.Join(fakeCgroupDir, "memory.stat")
+	maxPath := filepath.Join(fakeCgroupDir, "memory.max")
+	lowPath := filepath.Join(fakeCgroupDir, "memory.low")
+	currentPath := filepath.Join(fakeCgroupDir, "memory.current")
+	swapMaxPath := filepath.Join(fakeCgroupDir, "memory.swap.max")
+	numaStatPath := filepath.Join(fakeCgroupDir, "memory.numa_stat")
+	fakesub1CgroupDir := filepath.Join(fakeCgroupDir, "test1")
+	subnuma1StatPath := filepath.Join(fakesub1CgroupDir, "memory.numa_stat")
+	fakesub2CgroupDir := filepath.Join(fakeCgroupDir, "test2")
+	subnuma2StatPath := filepath.Join(fakesub2CgroupDir, "memory.numa_stat")
+
+	if err := os.WriteFile(statPath, []byte(memoryStatContents), 0o644); err != nil {
+		t.Fatal(err)
+	}
+	if err := os.WriteFile(maxPath, []byte(memoryMaxUsageContents), 0o644); err != nil {
+		t.Fatal(err)
+	}
+	if err := os.WriteFile(lowPath, []byte(memoryLimitContents), 0o644); err != nil {
+		t.Fatal(err)
+	}
+	if err := os.WriteFile(currentPath, []byte(memoryUsageContents), 0o644); err != nil {
+		t.Fatal(err)
+	}
+	if err := os.WriteFile(swapMaxPath, []byte(memoryMaxUsageContents), 0o644); err != nil {
+		t.Fatal(err)
+	}
+	if err := os.WriteFile(numaStatPath, []byte(memoryNUMAStatContents), 0o644); err != nil {
+		t.Fatal(err)
+	}
+	if err := os.MkdirAll(fakesub1CgroupDir, 0o755); err != nil {
+		t.Fatal(err)
+	}
+	if err := os.WriteFile(subnuma1StatPath, []byte(memoryNUMAStatContents), 0o644); err != nil {
+		t.Fatal(err)
+	}
+	if err := os.MkdirAll(fakesub2CgroupDir, 0o755); err != nil {
+		t.Fatal(err)
+	}
+	if err := os.WriteFile(subnuma2StatPath, []byte(memoryNUMAStatContents), 0o644); err != nil {
+		t.Fatal(err)
+	}
```

Comment:
You can also move `writeFileContents` to fscommon under some _test.go file and re-use it from here.